Packs/CommonScripts/Scripts/SearchIncidentsV2/SearchIncidentsV2.py

```diff
@@ -153,34 +155,48 @@ def search_incidents(args: Dict):   # pragma: no cover
         return 'Incidents not found.', {}, {}
 
     limit = arg_to_number(args.get('limit')) or DEFAULT_LIMIT
-    result_data_list = res[0]["Contents"]["data"]
+    all_found_incidents = res[0]["Contents"]["data"]
+    demisto.debug(
+        f'Amount of incidents before filtering = {len(all_found_incidents)} with args {args} before pagination'
+    )
+    all_found_incidents = add_incidents_link(apply_filters(all_found_incidents, args), platform)
+    demisto.debug(
+        f'Amount of incidents after filtering = {len(all_found_incidents)} before pagination'
+    )
     # adding 1 here because the default page number start from 0
     max_page = (res[0]["Contents"]["total"] // DEFAULT_PAGE_SIZE) + 1
+    demisto.debug(f'{max_page=}')
 
     page = STARTING_PAGE_NUMBER
-    while len(result_data_list) < limit and page < max_page:
+    while len(all_found_incidents) < limit and page < max_page:
```

Comment:
can we add <= maxpage here? Check this works

Reply:
it is possible, but not needed because paging starts from 0 and this is why we don't do <=, but do only <